### PR TITLE
fix issue apache/skywalking#3432

### PR DESCRIPTION
**Pull request by x22x22**
fix issue https://github.com/apache/skywalking/issues/3432

Before the repair: 

image

image

After repair:

image

image

trace span mock data:

[spans-mock.txt](https://github.com/apache/skywalking-rocketbot-ui/files/3586790/spans-mock.txt)

### CODE REVIEW

**Comment by TinyAllen** [comment]
@x22x22 did you use the ide format ？To many changed parts， l cant locate where is the core center to fix the bug ?

**Comment by x22x22** [comment]
> @x22x22 did you use the ide format ？To many changed parts， l cant locate where is the core center to fix the bug ?

yes, because the long code was placed on the same line, it feels unfriendly. But @wu-sheng asked me to modify it again, and I will mark the modified code for you.

**Comment by wu-sheng** [comment]
@x22x22 let's keep the style same as before in this PR. And discuss in 6.5.0, is that OK?

**Comment by x22x22** [comment]
> @x22x22 did you use the ide format ？To many changed parts， l cant locate where is the core center to fix the bug ?

@TinyAllen 

The fix is complete and the formatting is removed

**Comment by x22x22** [comment]
> @x22x22 did you use the ide format ？To many changed parts， l cant locate where is the core center to fix the bug ?


After repair:

image

mock data:
[spans-mock.txt](https://github.com/apache/skywalking-rocketbot-ui/files/3586791/spans-mock.txt)


You see the data marked with a yellow mark, the data does not need to be automatically created 1, 2, 3, 5. My original mock data is only 4,6,8, then 0 is created by me.

你看到用黄色记号标记出来的数据, 数据中不需要自动创建1,2,3,5. 我的原始mock数据里就只有4,6,8, 然后0是我创建出来的.

File: src/views/components/trace/trace-detail-chart-list.vue [hunk 1]
```diff
@@ -145,66 +146,111 @@ export default {
...
-            if (index === -1) {
-              for (let i = 0; i <= ref.parentSpanId; i += 1) {
-                fixSpans.push(
+        });
+        segmentHeaders.forEach((span) => {
+          if (span.refs.length) {
+            span.refs.forEach((ref) => {
+              const index = this.data.findIndex(i => (ref.parentSegmentId === i.segmentId && ref.parentSpanId === i.spanId));
+              if (index === -1) {
+                // create a known broken node.
```
Comment (wu-sheng) [inline src/views/components/trace/trace-detail-chart-list.vue]: @TinyAllen The following part makes sense to me. Create the missing parent vnode, then targeting its parent to root vnode. And every time check the vnode whether exist before doing `#push`.

**Pull request merged**